internal/api/drafts.go

```diff
@@ -695,6 +715,24 @@ func DraftsDocumentHandler(
 				productAbbreviation = p.Abbreviation
 			}
 
+			// Check if document is locked.
+			locked, err := hcd.IsLocked(docId, db, s)
+			if err != nil {
+				l.Error("error checking document locked status",
+					"error", err,
+					"method", r.Method,
+					"path", r.URL.Path,
+					"doc_id", docId,
+				)
+				http.Error(w, "Error getting document status", http.StatusNotFound)
+				return
+			}
+			// Don't continue if document is locked.
+			if locked {
+				http.Error(w, "Document is locked", http.StatusLocked)
```

Comment:
What happens in the frontend when we respond with this error? In some cases users may only notice that the document is locked when they change the state of the document. Do we have a good UX for them to identify what they should do to fix it? Can we point them to the sidebar where the locked message is shown?

cc: @jeffdaley

Reply:
Good questions! Currently this flash message will appear: 
<img width="383" alt="image" src="https://github.com/hashicorp-forge/hermes/assets/1489080/ee756c62-827f-40e8-bb4b-aa73f9aceeb0">

It's not an ideal UX, but is better than the current experience of when a doc gets into this state 🙂  We should probably at least handle this error in a way where the sidebar immediately changes to "locked mode" (this currently requires a browser refresh). As we discussed offline, we can handle these in a follow-up PR.